fiboa_cli/publish.py

```diff
@@ -0,0 +1,378 @@
+from vecorel_cli.create_stac import CreateStacCollection
+from vecorel_cli.validate import ValidateData
+
+from .registry import Registry
+
+STAC_EXTENSION = "https://stac-extensions.github.io/web-map-links/v1.2.0/schema.json"
+DESCRIPTIONS = {
+    "id": "Unique identifier",
+    "collection": "The collection identifier",
+    "inspire:id": "The INSPIRE identifier",
+    "determination:datetime": "Timestamp of the determination of the field boundary",
+    "metrics:area": "Field area in square meters",
+    "metrics:perimeter": "Field perimeter in square meters",
+    "crop:code_list": "A link to the code list",
+    "crop:code": "The crop code",
+    "crop:name": "Crop name in the original language",
+    "crop:name_en": "Crop name in English",
+    "hcat:name": "The machine-readable HCAT name of the crop",
+    "hcat:code": "The 10-digit HCAT code indicating the hierarchy of the crop",
+    "hcat:name_en": "The HCAT crop name translated into English",
+}
+
+
+class Publish(BaseCommand):
+    cmd_name = "publish"
+    cmd_help = f"Convert and publish a {Registry.project} dataset to source coop."
+    url_base = "https://data.source.coop/fiboa/data"
+
+    @staticmethod
+    def get_cli_args():
+        return {
+            **ConvertData.get_cli_args(),
+            "target": VECOREL_TARGET(folder=True),
+            "generate_meta": click.option(
+                "--generate-meta",
+                "-gm",
+                is_flag=True,
+                type=click.BOOL,
+                help="Generate README.txt and LICENSE.txt for the dataset if not present.",
+                default=False,
+            ),
+            "yes": click.option(
+                "--yes",
+                "-y",
+                is_flag=True,
+                type=click.BOOL,
+                help="Answer yes to all questions.",
+                default=False,
+            ),
+            "data_url": click.option(
+                "--data-url",
+                type=click.STRING,
+                help="When generating documentation, this is the link to the data.",
+            ),
+            "s3_upload_path": click.option(
+                "--s3-upload-path",
+                type=click.STRING,
+                help="Upload to this path on S3. By default it's the source coop fiboa data repository.",
+            ),
+        }
+
+    @staticmethod
+    def get_cli_callback(cmd):
+        def callback(dataset, *args, **kwargs):
+            return Publish(dataset).run(*args, **kwargs)
+
+        return callback
+
+    def __init__(self, dataset: str, data_url=None, s3_upload_path=None):
+        super().__init__()
+        self.cmd_title = f"Convert {dataset}"
+        self.dataset = dataset
+        self.data_url = data_url or f"{self.url_base}/{self.dataset}"
+        self.s3_upload_path = (
+            s3_upload_path or f"s3://us-west-2.opendata.source.coop/fiboa/data/{self.dataset}/"
+        )
+
+        if ConvertData.converters.is_converter(self.dataset):
```

Comment:
Can't you just initiate the Converters class instead of using the specific instance of ConvertData?

Reply:
Ah, yes. Will change it. The `Converters.is_converter()` is not the right check, as it checks for a filename... So I'll keep it short and direct!